rlba/environments/relu_logistic.py

```diff
@@ -0,0 +1,470 @@
+
+    # initialize the model and prameters
+    forward_partial = functools.partial(relu_logistic_fn, \
+                                        layer_dims=layer_dims,\
+                                        weight_var=weight_var,\
+                                        bias_var=bias_var, \
+                                        name=name)
+
+    # "pure model function"
+    model_fn = hk.transform(forward_partial)
+
+    # a dictionary recording the prior variances at different layers for weights
+    # and biases
+    prior_var_dict = {}
+    for i in range(1, len(layer_dims)):
+        prior_var_dict['W_'+str(i)] = weight_var/layer_dims[i-1]
+        prior_var_dict['b_'+str(i)] = bias_var
+
+    prior_var_dict['W_'+str(len(layer_dims))] = weight_var/layer_dims[-1]
+
+    @jax.jit
+    def loss_fn(params: dict, 
+                x: chex.Array, 
+                y: chex.Array, 
+                weights: Optional[Sequence[float]] = None, 
+                reg_params: dict = None) -> float:
+
+        """
+        Parameters:
+            params: dict
+                dictionary of weights and biases for the model
+            x: array_like
+                features
+            y: array_like
+                labels for features
+            weights: None or array_like
+                weights for weighing likelihood of observed data
+            reg_params: dictionary
+                values act as means to regularize against
+        """
+        
+        num_samples = len(y)
+
+        # compute the loss associated with the log likelihood
+        if num_samples > 0:
+            y_hat = model_fn.apply(params, None, x)
+            # vector encoding the negative log-likelihood
+            loss_vec = -jnp.multiply(y, jnp.log(y_hat))-\
+                jnp.multiply(1-y, jnp.log(1-y_hat))
+            # reweigh the vector by random weights
+            if weights is not None:
+                loss_vec = jnp.multiply(weights, loss_vec)
+            # average
+            loss_ll = jnp.mean(loss_vec)
+        else:
+            loss_ll = 0
+
+        # quadratic loss (from prior) for all weights and biases
+        loss_prior = 0
+        for key in prior_var_dict:
+            if reg_params is None:
+                weight_diff = params[name][key]
+            else:
+                weight_diff = params[name][key] - reg_params[name][key]
+                loss_prior = loss_prior + \
+                    jnp.sum(jnp.square(weight_diff))/(2*prior_var_dict[key])
+
+        # normalize the loss_prior based on number of samples,
+        # since the log-likelihood part is an average
+        loss_prior = loss_prior/num_samples
+        loss = loss_ll + loss_prior
+
+        return loss
+
+    grad_loss_fn = jax.jit(jax.value_and_grad(loss_fn, argnums=0))
+
+    return grad_loss_fn, model_fn
+
+def opt_pert_loss(hk_keys: hk.PRNGSequence, 
+                  grad_loss_fn, 
+                  model_fn, 
+                  num_epochs: int,
```

Comment:
to be consistent across the repo please use `n_XYZ` instead of `num_XYZs` in the new code. Fro example, use `n_epoch` instead of `num_epochs`.